libraries/shared/watcher/storage_watcher.go

```diff
@@ -91,7 +96,21 @@ func (watcher StorageWatcher) Execute() error {
 }
 
 func (watcher StorageWatcher) transformDiffs() error {
-	minID := 0
+	var minID int
+	if watcher.SkipOldDiffs {
+		mostRecentHeader, getHeaderErr := watcher.HeaderRepository.GetMostRecentHeader()
+		if getHeaderErr != nil {
+			return getHeaderErr
+		}
+		blockNumber := mostRecentHeader.BlockNumber - BlocksBackFromHead
+		diff, getDiffErr := watcher.StorageDiffRepository.GetFirstDiffForBlockHeight(blockNumber)
```

Comment:
Don't know if this makes any difference but maybe this function could just return the ID if that's all we want from the diff

Reply:
Good call - I had initially thought that this method could be useful in the future if it returned the whole diff, but we'll let future us deal with that, just in case it never is necessary.